### PR TITLE
Web socket

### PR DESCRIPTION
**Pull request by DmitryApan**
image

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1 @@
+[0204/205801.028:ERROR:directory_reader_win.cc(43)] FindFirstFile: ������� �� ������� ����� ��������� ����. (0x3)
```
Comment (siarhei-zharnasek) [inline debug.log]: what is it?

File: package.json [hunk 2]
```diff
@@ -50,12 +50,14 @@
     "react": "^16.13.1",
     "react-app-polyfill": "^0.2.2",
     "react-beautiful-dnd": "^13.0.0",
+    "react-bootstrap": "^1.4.3",
```
Comment (siarhei-zharnasek) [inline package.json]: do you need it?

File: package.json [hunk 3]
```diff
@@ -66,7 +68,8 @@
     "webpack": "4.28.3",
     "webpack-dev-server": "3.1.14",
     "webpack-manifest-plugin": "2.0.4",
-    "workbox-webpack-plugin": "3.6.3"
+    "workbox-webpack-plugin": "3.6.3",
```
Comment (siarhei-zharnasek) [inline package.json]: this one?

File: src/App.js [hunk 4]
```diff
@@ -6,22 +6,77 @@ import {
...
-        
+        super(props);        
+
         props.authorization();
     }
 
-    render() {     
+    state = {
+        socket: new WebSocket(urlWebSocket)
+    }
```
Comment (siarhei-zharnasek) [inline src/App.js]: semicolon

File: src/App.js [hunk 5]
```diff
@@ -6,22 +6,77 @@ import {
...
-        super(props);   
-        
+        super(props);        
+
         props.authorization();
     }
 
-    render() {     
+    state = {
+        socket: new WebSocket(urlWebSocket)
```
Comment (siarhei-zharnasek) [inline src/App.js:30]: comma missed

File: src/App.js [hunk 6]
```diff
@@ -6,22 +6,77 @@ import {
...
+            usersAppReceive, 
+            settingsReceive, 
+            userInfoReceive,
+            userInfo,
+            webSocket
+        } = this.props;
+
+        let obj = JSON.parse(data);
+        let field = obj.field;
+        let socketData = obj.data;
```
Comment (siarhei-zharnasek) [inline src/App.js]: please use desctructure

File: src/Card.js [hunk 7]
```diff
@@ -26,26 +26,27 @@ class Card extends React.Component {
 	}
 
 	render() {
-		const {_id, imageSrc, title, description} = this.props;
+		const {_id, imageSrc, title, description, cards} = this.props;
```
Comment (siarhei-zharnasek) [inline src/Card.js]: use 4 spaces or one tab here (not 8 spaces)

File: src/Data.js [hunk 8]
```diff
@@ -1,5 +1,7 @@
-let base = 'https://task-manager-by.herokuapp.com';
+// let base = 'https://task-manager-by.herokuapp.com';
+let base = 'http://localhost:8888';
 
+export let urlWebSocket = 'ws://localhost:8080';
```
Comment (siarhei-zharnasek) [inline src/Data.js:4]: I don't think it'll work on production

File: src/Data.js [hunk 9]
```diff
@@ -1,5 +1,7 @@
-let base = 'https://task-manager-by.herokuapp.com';
+// let base = 'https://task-manager-by.herokuapp.com';
```
Comment (siarhei-zharnasek) [inline src/Data.js:1]: why it's commented?

File: src/components/SettingsPanel/SettingsPanel.js [hunk 10]
```diff
@@ -35,11 +35,16 @@ export default function SettingsPanel({statuses, onDeleteStatus, onCreateStatus,
...
+        onEnableWebSocket(value);
+    }, [onEnableWebSocket]);
+
     return (
         <div className={styles.container}>
+            <Divider orientation="left">Statuses settings</Divider>
             {statuses.map((status, index) => (
                 <div key={index} className={styles.row}>
-                    <div className={styles.statusText} >
+                    <div className={styles.swichName} >
```
Comment (siarhei-zharnasek) [inline src/components/SettingsPanel/SettingsPanel.js]: switch

File: src/Data.js (hunk 8 above)
Comment (DmitryApan) [reply src/Data.js:4]: We don't have stable work WS on pordaction. 

I am running the backend locally

File: src/Data.js (hunk 9 above)
Comment (DmitryApan) [reply src/Data.js:1]: We were unable to start websocket. It was decided to run the backend locally on the computer

File: src/Data.js (hunk 8 above)
Comment (siarhei-zharnasek) [reply src/Data.js:4]: why it's not stable? did you add some code to backend?

File: src/Data.js (hunk 9 above)
Comment (siarhei-zharnasek) [reply src/Data.js:1]: yeah, now I remember

File: src/Data.js (hunk 8 above)
Comment (DmitryApan) [reply src/Data.js:4]: not. The web socket did not work for us. And you told me to run on the backend

File: src/Data.js (hunk 9 above)
Comment (DmitryApan) [reply src/Data.js:1]: do I leave it as it is?

Comment (siarhei-zharnasek) [reply src/Data.js:1]: please leave it

File: package.json (hunk 2 above)
Comment (siarhei-zharnasek) [reply package.json]: did you run `npm uninstall react-bootstrap` or just removed it yourself?

File: debug.log (hunk 1 above)
Comment (DmitryApan) [reply debug.log]: i don't know, i deleted it

File: package.json (hunk 2 above)
Comment (DmitryApan) [reply package.json]: just removed it

File: package.json (hunk 3 above)
Comment (DmitryApan) [reply package.json]: just removed it

File: src/App.js (hunk 4 above)
Comment (DmitryApan) [reply src/App.js]: fixed

File: src/App.js (hunk 5 above)
Comment (DmitryApan) [reply src/App.js:30]: I can't understand why where to put the comma

File: src/App.js (hunk 6 above)
Comment (DmitryApan) [reply src/App.js]: fixed

File: src/Card.js (hunk 7 above)
Comment (DmitryApan) [reply src/Card.js]: I checked, everything is fine in the code. Possibly a github bug

File: src/Data.js (hunk 9 above)
Comment (DmitryApan) [reply src/Data.js:1]: ok

File: src/components/SettingsPanel/SettingsPanel.js (hunk 10 above)
Comment (DmitryApan) [reply src/components/SettingsPanel/SettingsPanel.js]: fixed

**Review by siarhei-zharnasek — APPROVED** [review]

**Pull request merged**